src/app/homepage/homepage.component.ts

```diff
@@ -141,4 +142,20 @@ export class HomepageComponent implements OnInit, OnDestroy, AfterViewInit {
     scriptTag.id = '_carbonads_js';
     return scriptTag;
   }
+
+  createDocSearchScriptTag(): HTMLScriptElement {
+    const scriptTag = document.createElement("script");
+    scriptTag.type = "text/javascript";
+    scriptTag.src = "https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js";
+    scriptTag.async = true
+    scriptTag.onload = () => {
+      (window as any).docsearch({
+        apiKey: '9ea53de1a6911255834352bbbe4d3417',
```

Comment:
I don't know how secret this apiKey should be treated but is not better to have this in the `environment.ts` ?

Reply:
This secret is a searchOnly api key and is meant to be public. It is not sensitive at all. Most of website DocSearch featured does have this key within the DOM . You can find an example here with the bootstrap website:
[![image](https://user-images.githubusercontent.com/32097720/67022460-8fb86b00-f101-11e9-8de1-82af4edf3f49.png)](https://babeljs.io/)